### DISCUSSION TITLE
Next.js native routing stopped working

### DISCUSSION THREAD

**Discussion by mithatercan** [47ng/nuqs#381] [Q&A] [answered]
Hello,

I wanted to try using usequerystate in my project, so I installed it. However, after installing it, the native routing in Next.js stopped working, even though I reinstalled everything. None of the hooks usePathname, useRouter, and useSearchParams seem to work. Has anyone else experienced this issue or have an idea why it happened? I've tried everything, but none of the solutions have worked.

**Comment by franky47** [comment, marked as answer]
I see you closed the discussion, was the issue related with something else?

The library does patch the history API, but it forwards everything transparently, it should not impact other hooks, from Next.js or otherwise.

**Reply by mithatercan** [reply]
I had trouble understanding the issue and spent two days trying to debug it. Eventually, I decided to move all the code to a new project to resolve the problem. However, even after doing that, I still couldn't comprehend the issue. It seems to have started happening after I installed the library. Thank you for taking the time to respond.

**Reply by franky47** [reply]
Do you have a link to a reproducible example where this occurs? What's your Next.js version?
